### ISSUE TITLE
SFTP support

### ISSUE THREAD

**Issue by Tombstone2K** [files-community/Files#2407]
Please add SFTP support for Files-Beta.
I use KDE-Connect to access files on my android phone, adding SFTP client support would make life easier for many

**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**
Add any other context or screenshots about the feature request here. Ex: What device type/UI layout and app version?

**Comment by benjione** [comment]
I think this is a duplicate of #1629 and was merged with #1342 . However, FTP is not the same as SFTP. I suggest merging this with #1342 .

**Comment by ttmx** [comment]
This would be extremely nice and would remove the need for Filezilla and the likes, making it a much more uniform and convenient experience.

**Comment by yaichenbaum** [comment]
What are the requirements for this feature? How do you expect this feature to work.

**Comment by hez2010** [comment]
We can integrate SSH.NET and do what we did before for FTP.

**Comment by yaichenbaum** [comment]
Isn't our current implementation of FTP a little slow?

**Comment by hez2010** [comment]
> Isn't our current implementation of FTP a little slow

You're right. I'm expecting FluentFTP to use `IAsyncEnumerable` to list files from FTP in the future, instead of listing all entries at once.

**Comment by yaira2** [comment]
> We can integrate SSH.NET and do what we have done for FTP before. But we need to wait for the next release of SSH.NET for asynchronous IO support.

@hez2010 has this been released yet?

**Comment by hez2010** [comment]
> > We can integrate SSH.NET and do what we have done for FTP before. But we need to wait for the next release of SSH.NET for asynchronous IO support.
> 
> @hez2010 has this been released yet?

Unfortunately not yet. But I think we can use a custom internal fork.

**Comment by BLACK4585** [comment]
Is there any progress on this?

**Comment by itsWindows11** [comment]
I'll try looking into it. Is there a filesystem abstraction and stuff that I should do to properly implement everything as possible? I recall seeing many Win32 specific stuff in the codebase that don't seem to be abstracted away well?

cc @yaira2

**Comment by wildantea** [comment]
yeah, it would nice if sftp is supported. i found this app, for reference
https://apps.microsoft.com/detail/owlfiles-file-manager/9MSQ3SZK42KP?hl=en-us&gl=US
it has supported sftp.

**Comment by TimoNotThy** [comment]
@itsWindows11 how is this going?

**Comment by TimoNotThy** [comment]
@itsWindows11 How is this going?

**Comment by itsWindows11** [comment]
Time constraints on my end slowed down things, plus exams are coming up thus slowing down progress more in the next few weeks. I've got some of the basics ready already for SFTP though, will do a PR by the end of this month.

Sorry for any inconveniences caused here.

**Comment by TimoNotThy** [comment]
No worries, was just wondering if this was coming along or not

**Comment by Arlodotexe** [comment]
Files is using an older fork of the SecureFolderFs storage abstraction, which itself was a fork of the OwlCore.Storage project, an early adoption package for the CommunityToolkit.Storage proposal. I've worked with SecureFolderFS to incorporate their feedback and remove the need to maintain a separate fork, I would suggest we do the same for Files so we aren't duplicating planned work: https://github.com/Arlodotexe/OwlCore.Storage/issues/12

**Comment by pnoelfr** [comment]
It's GOOD ??

https://github.com/sshnet/SSH.NET/issues/1415

**Comment by 0x5bfa** [comment]
From @tmds, @itsWindows11 How do you think?

**Comment by itsWindows11** [comment]
@0x5bfa
https://github.com/files-community/Files/pull/15477#issuecomment-2395347057

**Comment by dedestem** [comment]
This will be such an great feature

**Comment by lakani** [comment]
is there is any updates on this

**Comment by Josh65-2201** [comment]
No
